### PR TITLE
Update for Beta 15

### PR DESCRIPTION
**Pull request by askvortsov1**
- Beta 15 upgrade guide
- Update to use new extenders
- Document api throttling
- Document service provider layer
- Add documentation of API request lifecycle
- Document `parse` and `render` for formatter
- Update recommended extension version constraints
- Add blank pages for model slugging @tankerkiller125  and admin UX @KyrneDev

### CODE REVIEW

File: docs/extend/api-throttling.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+:::
+
+## Custom Throttlers
+
+The format for a custom throttler is extremely simple: all you need is a closure or invokable class that takes the current request as an argument, and returns one of:
+
+- `false`: This explicitly bypasses throttling for this request, overriding all other throttlers
+- `true`: This marks the request as to be throttled.
+
+Any other outputs will be ignored. If the throttler allows a given request, it should probably return `null` (or not return anything), not `true`.
```
Comment (tankerkiller125) [inline docs/extend/api-throttling.md:18]: Why not just take the note/comment about null and make it a 3rd response?

**Review by clarkwinkelmann — CHANGES_REQUESTED** [review]

File: docs/extend/data.md [hunk 2]
```diff
@@ -4,12 +4,24 @@ Data is the foundation of any forum, so you're going to need to play nice with i
...
+2. The request is run through [middleware](middleware.md), and routed to the proper controller. You can learn more about controllers as a whole on our [routes and content documentation](routes.md). Assuming the request is to the API (which is the case for this section), the controller that handles the request will be a subclass of `Flarum\Api\AbstractSerializeController`.
+3. Any modifications done by extensions to the controller via the [`ApiController` extender](#extending-api-controllers) are applied. This could entail changing sort, adding includes, changing the serializer, etc.
+4. The `$this->data()` method of the controller is called, yielding some raw data that should be returned to the client. Typically, this data will take the form of a Laravel Eloquent model collectio or instance, which has been retrieved from the database. Each controller is responsible for implementing its own `data` method. Note that for `PATCH`, `POST`, and `DELETE` requests, `data` will perform the operation in question, and return the modified model instance.
```
[the remark this answers is not in the document]
Comment (SychO9) [reply docs/extend/data.md:15]: also the data can be a simple php array

File: docs/extend/data.md [hunk 3]
```diff
@@ -4,12 +4,24 @@ Data is the foundation of any forum, so you're going to need to play nice with i
...
-::: warning
-Many of the APIs described on this page are planned to change in the near future.
-:::
+
+## API Request Lifecycle
+
+Before we go into detail about how to extend Flarum's data API, it's worth thinking about the lifecycle of a typical data request:
+
+1. An HTTP request is sent to Flarum's API. Typically, this will come from the Flarum frontend, but external programs can also interact with the API. Flarum's API mostly follows the [JSON:API](https://jsonapi.org/) specification, so accordingly, requests should follow [said specification](https://jsonapi.org/format/#fetching).
+2. The request is run through [middleware](middleware.md), and routed to the proper controller. You can learn more about controllers as a whole on our [routes and content documentation](routes.md). Assuming the request is to the API (which is the case for this section), the controller that handles the request will be a subclass of `Flarum\Api\AbstractSerializeController`.
```
[the remark this answers is not in the document]
Comment (askvortsov1) [reply docs/extend/data.md:13]: Plural of middleware is middleware

File: docs/extend/data.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (askvortsov1) [reply docs/extend/data.md:15]: > Is the method really called data for each type of controller?

Yep!

File: docs/extend/data.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (clarkwinkelmann) [reply docs/extend/data.md:13]: Didn't know that :hankey: 

Still sounds confusing to me. What about "the middleware pipe" then?

**Review by clarkwinkelmann — APPROVED** [review]
I added a suggestion in reply to my previous review. Otherwise all good!

**Pull request merged**